ingest.py

```diff
@@ -1,28 +1,53 @@
 """Load html from files, clean up, split, ingest into Weaviate."""
-import pickle
+from bs4 import BeautifulSoup as Soup
+import weaviate
+import os
 
-from langchain.document_loaders import ReadTheDocsLoader
-from langchain.embeddings import OpenAIEmbeddings
+from langchain.document_loaders.recursive_url_loader import RecursiveUrlLoader
+from langchain.embeddings import OpenAIEmbeddings, CohereEmbeddings
+from langchain.schema import Document
 from langchain.text_splitter import RecursiveCharacterTextSplitter
-from langchain.vectorstores.faiss import FAISS
+from langchain.vectorstores import Weaviate
 
+WEAVIATE_URL=os.environ["WEAVIATE_URL"]
+WEAVIATE_API_KEY=os.environ["WEAVIATE_API_KEY"]
 
 def ingest_docs():
     """Get documents from web pages."""
-    loader = ReadTheDocsLoader("langchain.readthedocs.io/en/latest/")
-    raw_documents = loader.load()
-    text_splitter = RecursiveCharacterTextSplitter(
-        chunk_size=1000,
-        chunk_overlap=200,
-    )
-    documents = text_splitter.split_documents(raw_documents)
-    embeddings = OpenAIEmbeddings()
-    vectorstore = FAISS.from_documents(documents, embeddings)
 
-    # Save vectorstore
-    with open("vectorstore.pkl", "wb") as f:
-        pickle.dump(vectorstore, f)
+    urls = [
+        "https://api.python.langchain.com/en/latest/api_reference.html#module-langchain",
+        "https://python.langchain.com/docs/get_started", 
+        "https://python.langchain.com/docs/modules", 
+        "https://python.langchain.com/docs/guides",
+        "https://python.langchain.com/docs/ecosystem",
+        "https://python.langchain.com/docs/additional_resources",
+    ]
 
+    documents = []
+    for url in urls:
+        loader = RecursiveUrlLoader(url=url, max_depth=1 if url == urls[0] else 3, extractor=lambda x: Soup(x, "lxml").text, prevent_outside=False)
```

Comment:
we probably want to use the HtmlToText transformer?

Reply:
i think you want to set `prevent_outside=True` and then set a high max depth

Otherwise you could have the loader for `guides` loading things from `modules` (since guides links to modules)